build-logic/packaging/src/main/kotlin/gradlebuild.public-api-jar.gradle.kts

```diff
@@ -54,37 +45,29 @@ val externalRuntimeClasspath = configurations.resolvable("externalRuntimeClasspa
     configureAsRuntimeJarClasspath(objects)
 }
 
-enum class Filtering {
-    PUBLIC_API, ALL
+// Defines configurations used to resolve the public Gradle API.
+val distribution = configurations.dependencyScope("distribution") {
+    description = "Dependencies to extract the public Gradle API from"
 }
-
-val filteredAttribute: Attribute<Filtering> = Attribute.of("org.gradle.apijar.filtered", Filtering::class.java)
-
-dependencies {
-    artifactTypes.getByName("jar") {
-        attributes.attribute(filteredAttribute, Filtering.ALL)
-    }
-
-    // Filter and shrink the published API.
-    registerTransform(ShrinkPublicApiClassesTransform::class.java) {
-        from.attribute(filteredAttribute, Filtering.ALL)
-            .attribute(LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE, objects.named(LibraryElements::class.java, LibraryElements.JAR))
-        to.attribute(filteredAttribute, Filtering.PUBLIC_API)
+val distributionClasspath = configurations.resolvable("distributionClasspath") {
+    extendsFrom(distribution.get())
+    attributes {
+        attribute(ImplementationCompletenessAttribute.attribute, ImplementationCompletenessAttribute.STUBS)
     }
 }
 
 val task = tasks.register<Jar>("jarGradleApi") {
     from(distributionClasspath.map { configuration ->
         configuration.incoming.artifactView {
-            attributes {
-                attribute(filteredAttribute, Filtering.PUBLIC_API)
-            }
             componentFilter { componentId -> componentId is ProjectComponentIdentifier }
         }.files
-    })
+    }) {
+        // TODO Use better filtering
+        include("**/*.class")
+    }
```

Comment:
🤔 The better filtering deserves a follow-up issue.

Reply:
https://github.com/gradle/gradle/issues/32138